jobsdb/jobsdb.go

```diff
@@ -1450,14 +1447,22 @@ func (jd *HandleT) constructParameterJSONQuery(table string, parameterFilters []
 	return fmt.Sprintf(`(%s.parameters @> '{%s}' %s)`, table, strings.Join(allKeyValues, ","), opQuery)
 }
 
+type cacheValue int
+
+const (
+	mayFindJobs cacheValue = iota
+	noJobs
+	willTryToSet
+)
+
 /*
 * If a query returns empty result for a specific dataset, we cache that so that
 * future queries don't have to hit the DB.
```

Comment:
Update the comment. Also, add some comment on why we needed to have willTryToSet. If not anything, copy paste the slack message here

Reply:
Sure.